util/omrutil/j9memclr.cpp

```diff
@@ -113,23 +120,15 @@ OMRZeroMemory(void *ptr, uintptr_t length)
 	memset(ptr, 0, (size_t)length);
 #elif defined(J9ZOS390)
 
-	if (((struct IHAPSA *)0)->FLCFGIEF) {
+	if (useJ9zerz10Assembly && ((struct IHAPSA *)0)->FLCFGIEF) {
 		J9ZERZ10(ptr, length);
 	} else {
 		memset(ptr, 0, (size_t)length);
 	}
 #else
 #if (defined (LINUX) && defined(S390)) && !defined(OMRZTPF)
```

Comment:
While you're here, might as well tidy up this ifdef:

```c++
#else
#if (defined (LINUX) && defined(S390)) && !defined(OMRZTPF)
```

to something like:
```c++
#elif defined(LINUX) && defined(S390) && !defined(OMRZTPF)
```

(note that this is removing a level of nesting)